This doesn't work as expected:

─➤  gmake test
cd test;\
SQUEAK_PLUGINS=/export/home/andreas/oi-userland/components/runtime/smalltalk/squeak/build/prototype/i386//usr/lib/squeak/4.19.15-3822;\
export SQUEAK_PLUGINS;\
../inisqueak4 -n;\
/export/home/andreas/oi-userland/components/runtime/smalltalk/squeak/build/prototype/i386//usr/lib/squeak/4.19.15-3822/squeakvm squeak.image testrunner.st
Downloading SqueakV46.sources.zip from http://files.squeak.org/4.6/
--2022-10-31 19:22:55--  http://files.squeak.org/4.6//SqueakV46.sources.zip
Resolving files.squeak.org (files.squeak.org)... 116.203.28.174
Connecting to files.squeak.org (files.squeak.org)|116.203.28.174|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6406218 (6.1M) [application/zip]
Saving to: ‘SqueakV46.sources.zip’

SqueakV46.sources.zip                      100%[========================================================================================>]   6.11M  8.98MB/s    in 0.7s    

2022-10-31 19:22:55 (8.98 MB/s) - ‘SqueakV46.sources.zip’ saved [6406218/6406218]

Archive:  SqueakV46.sources.zip
replace SqueakV46.sources? [y]es, [n]o, [A]ll, [N]one, [r]ename: y
  inflating: SqueakV46.sources       
Downloading Squeak4.6-15118.zip from http://files.squeak.org/4.6/
--2022-10-31 19:23:04--  http://files.squeak.org/4.6//Squeak4.6-15118.zip
Resolving files.squeak.org (files.squeak.org)... 116.203.28.174
Connecting to files.squeak.org (files.squeak.org)|116.203.28.174|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 8303401 (7.9M) [application/zip]
Saving to: ‘Squeak4.6-15118.zip’

Squeak4.6-15118.zip                        100%[========================================================================================>]   7.92M  11.5MB/s    in 0.7s    

2022-10-31 19:23:05 (11.5 MB/s) - ‘Squeak4.6-15118.zip’ saved [8303401/8303401]

Archive:  Squeak4.6-15118.zip
replace Squeak4.6-15118.changes? [y]es, [n]o, [A]ll, [N]one, [r]ename: y
  inflating: Squeak4.6-15118.changes  
replace Squeak4.6-15118.image? [y]es, [n]o, [A]ll, [N]one, [r]ename: y
  inflating: Squeak4.6-15118.image   
Copying to default image name ..
Squeak is already installed in this directory
/bin/bash: line 5: /export/home/andreas/oi-userland/components/runtime/smalltalk/squeak/build/prototype/i386//usr/lib/squeak/4.19.15-3822/squeakvm: No such file or directory
gmake: *** [Makefile:83: SUnitRunAll] Error 127

--- a/components/runtime/smalltalk/squeak/Makefile
+++ b/components/runtime/smalltalk/squeak/Makefile
@@ -55,7 +55,7 @@ include ../../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME=		squeak
 COMPONENT_VERSION=	4.19.15
-COMPONENT_REVISION=	1
+COMPONENT_REVISION=	2
 COMPONENT_SUMMARY=	The Squeak Virtual Machine
 COMPONENT_PROJECT_URL=	http://www.squeak.org
 COMPONENT_ARCHIVE_URL=  http://squeakvm.org
@@ -71,7 +71,10 @@ SVN_HASH=  sha256:5b8cbf212bc6913c25699cd4e22dbd1ee8778deb5130b07c99033ede1b46ad
 COMPONENT_LICENSE=      Squeak4
 COMPONENT_LICENSE_FILE= squeak4.license
 
-TEST_TARGET= $(NO_TESTS)
+# the tests are ran only on 32bit for Squeak classical VM
+# because inisqueak downloads a 32 bit image (provided by squeak.org)
+# for OpenSmalltalk (see cog-spur and stack-spur) 32bit + 64bit
+COMPONENT_TEST_CMD=	$(COMPONENT_TEST_RESULTS_DIR)/testrunner.sh $(BITS) $(COMPONENT_DIR) $(BUILD_DIR_32)
 
 include $(WS_MAKE_RULES)/common.mk
 

--- a/components/runtime/smalltalk/squeak/test/results-all.master
+++ b/components/runtime/smalltalk/squeak/test/results-all.master
@@ -1,0 +1,22 @@
+SUnit Results
+Squeak4.6
+solaris2.11
+Squeak4.6 of 21 July 2021 [latest update: #15118]
+Failed Tests
+'BrowserTest>>#testSelectClassNamedPreservesPlace'
+'ClosureCompilerTest>>#testSourceRangeAccessForBlueBookInjectInto'
+'DecompilerTests>>#testDecompilerInClassesPNtoPZ'
+'ExceptionTests>>#testHandlerFromAction'
+'FontTest>>#testParagraph'
+'FontTest>>#testParagraphFallback'
+'PackageDependencyTest>>#testMonticello'
+'PackageDependencyTest>>#testNetwork'
+'PackageDependencyTest>>#testPreferenceBrowser'
+'SocketTest>>#testSocketReuse'
+'SocketTest>>#testUDP'
+'TestVMStatistics>>#testVmStatisticsReportString'
+'TestValueWithinFix>>#testValueWithinTimingRepeat'
+Errors
+Total Number of Passed Tests: 3750
+Total Number of Failures: 13
+Total Number of Errors: 0

--- a/components/runtime/smalltalk/squeak/test/testrunner.sh
+++ b/components/runtime/smalltalk/squeak/test/testrunner.sh
@@ -1,0 +1,46 @@
+#!/bin/sh
+
+BITS=$1
+COMPONENT_DIR=$2
+BUILD_DIR=$3
+TEST_TARGET=$4
+
+LOGFILE=testrunner-log.$BITS
+
+# the tests could be ran after pkg install as:
+#       inisqueak -n;squeak squeak.image testrunner.st
+# or interactively simply by opening a Squeak image and going to TestRunner
+
+# however here we want to do this on the newly built 32bit VM
+# so we start squeak from the BUILD_DIR directory
+# we only test the 32bit VM because inisqueak downloads a 32bit image
+
+# WARNING HACK : fake 64bit results because we don't want to run 64bit test
+case $BITS in
+ 32) ;;
+ 64) cat $COMPONENT_DIR/test/results-all.master ;exit 0;;
+  *) echo "Unknown BITS $BITS";exit 1;;
+esac
+
+# unlike OpenSmalltalk libtool/configure plugins are not found in build dir
+# make sure that the newly built VM loads plugins fro the build dir
+
+for plugin in vm-display-X11 vm-display-null vm-sound-pulse vm-sound-null UUIDPlugin SqueakSSL
+do
+  SQUEAK_PLUGINS=$SQUEAK_PLUGINS:$BUILD_DIR/$plugin
+# LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$BUILD_DIR/$plugin
+done
+#echo "Setting LD_LIBRARY_PATH to $LD_LIBRARY_PATH" >> $LOGFILE
+#export LD_LIBRARY_PATH
+echo "Setting SQUEAK_PLUGINS to $SQUEAK_PLUGINS" >> $LOGFILE
+export SQUEAK_PLUGINS
+
+# download squeak.image to current directory
+$COMPONENT_DIR/inisqueak4 -n >> $LOGFILE 2>&1
+
+# start squeak binary from the build dir
+$BUILD_DIR/squeakvm squeak.image $COMPONENT_DIR/test/testrunner.st >> $LOGFILE 2>&1
+
+# testrunner.st saves output in a file , dump that file as output
+cat results-32.vm
+

--- a/components/runtime/smalltalk/squeak/test/testrunner.st
+++ b/components/runtime/smalltalk/squeak/test/testrunner.st
@@ -1,0 +1,27 @@
+| file result testRunner |
+Utilities setAuthorInitials:'hipster'.
+testRunner _ TestRunner new runAll.
+result _ testRunner result.
+file _ StandardFileStream fileNamed:'results-32.vm'.
+file nextPutAll: 'SUnit Results'; lf.
+file nextPutAll: Smalltalk version; lf.
+file nextPutAll: Smalltalk osVersion; lf.
+file nextPutAll: Smalltalk vmVersion; lf.
+file nextPutAll: 'Failed Tests'; lf.
+testRunner failedList do:
+		[ : each | each printOn:file.  file lf ].
+file nextPutAll: 'Errors'; lf.
+testRunner errorList do:
+		[ : each | each printOn:file.  file lf ].
+file nextPutAll: 'Total Number of Passed Tests: '.
+result passedCount printOn:file.
+file lf.
+file nextPutAll: 'Total Number of Failures: '.
+result failureCount printOn:file.
+file lf.
+file nextPutAll: 'Total Number of Errors: '.
+result errorCount printOn:file.
+file lf.
+file close.
+Smalltalk quitPrimitive.
+

